# Add missing cstring header include statements
# 
# Davide Vanzo (Vanderbilt University)
diff -ru tandem-linux-17-02-01-4.orig/src/base64.h tandem-linux-17-02-01-4/src/base64.h
--- tandem-linux-17-02-01-4.orig/src/base64.h	2019-06-27 11:06:29.621489466 -0500
+++ tandem-linux-17-02-01-4/src/base64.h	2019-06-27 11:37:16.017540557 -0500
@@ -1,5 +1,7 @@
 #ifndef BASE64_H
 #define BASE64_H
 
+#include <string.h>
+
 int b64_decode_mio (char *dest, char *src, size_t size);
 #endif /* BASE64_H */
diff -ru tandem-linux-17-02-01-4.orig/src/mcleave.h tandem-linux-17-02-01-4/src/mcleave.h
--- tandem-linux-17-02-01-4.orig/src/mcleave.h	2019-06-27 11:06:29.621489466 -0500
+++ tandem-linux-17-02-01-4/src/mcleave.h	2019-06-27 11:27:02.589523583 -0500
@@ -137,6 +137,8 @@
  * and rapidly test a peptide sequence to see if it is cleavable.
  * NOTE: mcleave.h does not have a corresponding .cpp file
  */
+#include <cstring>
+
 class mcleave_single
 {
 public:
diff -ru tandem-linux-17-02-01-4.orig/src/mhistogram.h tandem-linux-17-02-01-4/src/mhistogram.h
--- tandem-linux-17-02-01-4.orig/src/mhistogram.h	2019-06-27 11:06:29.621489466 -0500
+++ tandem-linux-17-02-01-4/src/mhistogram.h	2019-06-27 11:21:02.141513609 -0500
@@ -146,6 +146,7 @@
  * mhistogram is included in the mspectrum.h file only
  */
 #include <climits>
+#include <cstring>
 
 class mhistogram	{
 public:
diff -ru tandem-linux-17-02-01-4.orig/src/mmotif.h tandem-linux-17-02-01-4/src/mmotif.h
--- tandem-linux-17-02-01-4.orig/src/mmotif.h	2019-06-27 11:06:29.621489466 -0500
+++ tandem-linux-17-02-01-4/src/mmotif.h	2019-06-27 11:24:42.637519710 -0500
@@ -137,6 +137,7 @@
  * mmotif is a specialty class meant to store information about protein sequence motifs.
  * NOTE: mmotif.h does not have a corresponding .cpp file
  */
+#include <cstring>
 
 class mmotifres
 {
diff -ru tandem-linux-17-02-01-4.orig/src/msequenceserver.h tandem-linux-17-02-01-4/src/msequenceserver.h
--- tandem-linux-17-02-01-4.orig/src/msequenceserver.h	2019-06-27 11:06:29.621489466 -0500
+++ tandem-linux-17-02-01-4/src/msequenceserver.h	2019-06-27 11:35:41.169537932 -0500
@@ -140,6 +140,9 @@
  * deque of file names is then extracted and the process continues until the last
  * sequence from the last file in the deque is read into the msequencecontainer.
  */
+
+#include <cstring>
+
 typedef map<string,size_t> DESMAP;
 
 class msequenceCollection;
